code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +155,112 @@
+			spawn_group_size += max(0, group_job.current_positions * group_job.dynamic_spawn_group_multiplier - group_job.total_position_delta)
+		// If we are sharing a proxy, both proxies need to be active
+		if (min_pop_redirect != null && min_pop_redirect == group_job.min_pop_redirect && SSjob.initial_players_to_assign < group_job.min_pop && SSjob.initial_players_to_assign < min_pop)
+			spawn_group_size += max(0, group_job.current_positions * group_job.dynamic_spawn_group_multiplier - group_job.total_position_delta)
+	return spawn_group_size - total_position_delta
+
+/// Get the number of positions that are available for this job.
+/// Will typically return the total_positions value with the delta set by the HOP added on.
+/// If the min/max pop is not met returns 0.
+/// If the total positions is -1, returns -1 representing an unlimited position count
+/// If a dynamic spawn group is set, or the min pop is not met and the dynamic spawn group of the min_pop_redirect
+/// role is set, then it will compute the number of positions available based off of how populated the other jobs
+/// in the group are.
+/// Certain roles such as geneticist will return the smallest the smaller value between the number of jobs remaining
+/// in the medical department, and the max limit of geneticist slots available, if the lowpop limit has not been met.
+/// = Params =
+/// ignore_self_limit: bool => If set, then the max limit of the job will be ignored when using the dynamic group
+/// scaling calculations.
+/datum/job/proc/get_spawn_position_count(ignore_self_limit = FALSE)
+	var/player_count = SSjob.initial_players_to_assign
+	// SSjob has not been allocated yet
+	if (!player_count)
+		player_count = length(GLOB.clients)
+	// Out of range, and no proxy
+	if (player_count < min_pop && !min_pop_redirect)
+		return 0
+	if (player_count > max_pop)
+		return 0
+	// Unlimited, though we are limited if we use a dynamic spawn group
+	// We must be:
+	// - Unlimited ourselves
+	// - Not present in a dynamic spawn group
+	// - Not using a proxy due to lowpop, or the proxy has no dynamic spawn group
+	if (total_positions == -1 && !dynamic_spawn_group && (player_count >= min_pop || !min_pop_redirect || !min_pop_redirect::dynamic_spawn_group))
+		return -1
+	// If the population is lower than our min pop spawn amount
+	// then we will instead treat ourselves
+	var/datum/job/proxy = src
+	if (min_pop_redirect && player_count < min_pop)
+		proxy = SSjob.GetJob(min_pop_redirect::title)
+	// Does not have a spawn group
+	if (!proxy.dynamic_spawn_group)
+		// The proxy role allows for infinite joining, so we do too
+		if (proxy.total_positions == -1)
+			return -1
+		return max(proxy.total_positions + total_position_delta, 0)
+	// Calculate spawn group size
+	var/spawn_group_minimum = INFINITY
+	for (var/datum/job/other in SSjob.occupations)
+		// Find everything in the same group, doesn't matter if its us
+		if (other.dynamic_spawn_group != proxy.dynamic_spawn_group)
+			continue
```

Comment:
I think this "for loop" is heavy. I think you should cache the grouped jobs into SSdepartment. (or SSjob)

Reply:
No changes made: Caching group jobs would not provide a meaningful performance improvement as it is relatively light compared to the rest of the loop. The only way would be to reactively track the size of the group which increases the complexity of the code significantly while providing negligible real performance improvements (as there isn't a performance bottleneck to begin with).

Reply:
and then I think it is fine as is until we have more groups.